### PR TITLE
GSN support

### LINKED ISSUE #56: Merge the GSN Recipient base contracts from vanilla Contracts into the Ethereum Package
Merge the changes from https://github.com/OpenZeppelin/openzeppelin-contracts/issues/1843 into this package.

### PR DESCRIPTION
**Pull request by spalladino**
Fixes #56.

Adds GSN support from the vanilla repository.

### CODE REVIEW

**Comment by spalladino** [comment]
Pending: update test.sh and use latest relayer binary

File: contracts/mocks/GSNBouncerERC20FeeMock.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
+pragma solidity ^0.5.0;
+
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "../GSN/GSNRecipient.sol";
+import "../GSN/bouncers/GSNBouncerERC20Fee.sol";
+
+contract GSNBouncerERC20FeeMock is Initializable, GSNRecipient, GSNBouncerERC20Fee {
+    function initialize(string memory name, string memory symbol, uint8 decimals) initializer public {
```
Comment (nventuro) [inline contracts/mocks/GSNBouncerERC20FeeMock.sol]: We should turn this into a constructor

File: contracts/mocks/GSNBouncerSignatureMock.sol [hunk 2]
```diff
@@ -0,0 +1,18 @@
+pragma solidity ^0.5.0;
+
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "../GSN/GSNRecipient.sol";
+import "../GSN/bouncers/GSNBouncerSignature.sol";
+
+contract GSNBouncerSignatureMock is Initializable, GSNRecipient, GSNBouncerSignature {
+    function initialize(address trustedSigner) initializer public {
```
Comment (nventuro) [inline contracts/mocks/GSNBouncerSignatureMock.sol]: Also turn into constructor

File: contracts/mocks/GSNContextMock.sol [hunk 3]
```diff
@@ -0,0 +1,46 @@
+pragma solidity ^0.5.0;
+
+import "./ContextMock.sol";
+import "../GSN/GSNContext.sol";
+import "../GSN/IRelayRecipient.sol";
+
+// By inheriting from GSNContext, Context's internal functions are overridden automatically
+contract GSNContextMock is ContextMock, GSNContext, IRelayRecipient {
```
Comment (nventuro) [inline contracts/mocks/GSNContextMock.sol]: This contract is not initializing its parents (`GSNContext`)

File: contracts/mocks/GSNRecipientMock.sol [hunk 4]
```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.5.0;
+
+import "../GSN/GSNRecipient.sol";
+
+contract GSNRecipientMock is GSNRecipient {
```
Comment (nventuro) [inline contracts/mocks/GSNRecipientMock.sol]: Not initializing `GSNRecipient`

**Pull request merged**